### PR TITLE
6 adding authentication

### PR DESCRIPTION
**Pull request by AnaLanda**

### CODE REVIEW

**Review by jah701 — APPROVED** [review]
top

File: src/main/java/com/internet/shop/controllers/user/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String pwd = req.getParameter("pwd");
+        try {
+            User user = authenticationService.login(login, pwd);
+            HttpSession session = req.getSession();
+            session.setAttribute("user_id", user.getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/user/LoginController.java]: let's make user_id a constant

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Override
+    public User login(String login, String password)
+            throws AuthenticationException {
+        User userFromDB = userService.findByLogin(login).orElseThrow(() ->
+                new AuthenticationException("Incorrect user name or password."));
+        if (userFromDB.getPassword().equals(password)) {
+            return userFromDB;
+        }
+        throw new AuthenticationException("Incorrect user name or password.");
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: either make both checks in one if() or make exception message a constant

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+public class AuthenticationFilter implements Filter {
+    private static final String USER_ID = "user_id";
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]:
```suggestion
```

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        if (userId == null || userService.getById(userId) == null) {
+            resp.sendRedirect("/login");
+            return;
+        }
+        filterChain.doFilter(req, resp);
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]:
```suggestion
```

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**